requests/download_tpdi_deliveries.py

```diff
@@ -0,0 +1,102 @@
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+from pathlib import Path
+from typing import Any
+
+# Sentinel Hub
+from sentinelhub import (
+    SHConfig,
+)
+
+
+def sentinelhub_compliance_hook(response):
+    response.raise_for_status()
+    return response
+
+
+# Configuration of the access to SentinelHub
+config = SHConfig()
+oauth = config.instance_id
+
+# Your client credentials
+client_id = config.sh_client_id
+client_secret = config.sh_client_secret
+
+# Create a session
+client = BackendApplicationClient(client_id=client_id)
+oauth = OAuth2Session(client=client)
+
+# Get token for the session
+token = oauth.fetch_token(
+    token_url="https://services.sentinel-hub.com/oauth/token",
+    client_secret=client_secret,
+)
+
+# All requests using this session will have an access token automatically added
+resp = oauth.get("https://services.sentinel-hub.com/oauth/tokeninfo")
+oauth.register_compliance_hook("access_token_response", sentinelhub_compliance_hook)
+
+subscription_id = "<your subscription id>"
+order_id ="<your order id>"
+
+
+def download_file(url: str, output_path: str, overwrite: bool=False) -> None:
+    """Download Third Party Data Import file
+    :param url: Url pointing to a file.
+    :param output_path: Local path to save the file.
+    :param (optional): Overwrite a file or not.
+    """
+    p = Path(output_path)
+    if not overwrite and p.is_file():
+        return
+    p.parent.mkdir(parents=True, exist_ok=True)
+    response = oauth.get(url)
+    response.raise_for_status()
+    with open(output_path, "wb") as f:
+        f.write(response.content)
+    print(f"Downloaded file: {output_path}")
+
+
+def download_deliveries(base_url: str, deliveries: dict[str, Any], overwrite: bool=False) -> None:
+    """Download all deliveries
+    :param base_url: Url pointing to deliveries.
+    :param deliveries: Response of get subscription/order deliveries.
+    :param (optional): Overwrite a file or not.
+    """
+    for delivery in deliveries["data"]:
+        delivery_id = delivery["id"]
+        files_url = base_url + f"/{delivery_id}/files"
+        files_req = oauth.get(files_url)
+        for file in files_req.json():
+            download_file(files_url + f"/{file}", f"./{delivery_id}/{file}", overwrite)
```

Comment:
It would be nice to have the subscription ID or the order ID as the top level folder. So `./{subscription/order_id}/{delivery_id}/{file}`. So if you download files from multiple subscriptions you can still tell aparat where which delivery belongs to.

Reply:
Make sense. I updated the function.